File: StudyEnglishJava/src/main/java/org/example/studyenglishjava/dto/IWordStatus.java
package org.example.studyenglishjava.dto;

public sealed interface IWordStatus permits IWordStatus.Learned, IWordStatus.Blurred, IWordStatus.Unknown {

    int value();

    final class Learned implements IWordStatus {
        @Override
        public int value() {
            return 1;
        }

    }

    final class Blurred implements IWordStatus {
        @Override
        public int value() {
            return 2;
        }
    }

    final class Unknown implements IWordStatus {
        @Override
        public int value() {
            return 3;
        }
    }
}

File: LearningEnglish/app/src/main/java/com/example/learningenglish/word/model/IWordStatus.kt
package com.example.learningenglish.word.model

sealed interface IWordStatus {

    fun value(): Int

    data object Learned : IWordStatus {
        override fun value(): Int {
            return 1
        }
    }

    data object Blurred : IWordStatus {
        override fun value(): Int {
            return 2
        }
    }

    data object Unknown : IWordStatus {
        override fun value(): Int {
            return 3
        }
    }

}